ultraplot/tests/test_sharing.py

```diff
@@ -0,0 +1,98 @@
+import pytest, ultraplot as uplt
+
+"""
+Sharing levels for subplots determine the visibility of the axis labels and tick labels.
+
+Axis labels are pushed to the border subplots when the sharing level is greater than 1.
+
+Ticks are visible only on the border plots when the sharing level is greater than 2.
+
+Or more verbosely:
+    sharey = 0: no sharing, all labels and ticks visible
+    sharey = 1: share axis, all labels and ticks visible
+    sharey = 2: share limits
+    sharey = 3 or True, share both ticks and labels
+A similar story holds for sharex.
```

Comment:
This comment should specify when limits are shared for all levels. Also IDK what "share axis" means here.